### PR TITLE
[incubator/elasticsearch] Add customization of elasticsearch.yml

### PR DESCRIPTION
**Pull request by Emmenemoi**
Add customization of elasticsearch.yml using values.yml : cluster.name and cluster.config

### CODE REVIEW

**Comment by Emmenemoi** [comment]
@simonswine Could you please check if it is ok to merge? Very simple addition but very useful.

File: incubator/elasticsearch/templates/elasticsearch-configmap.yaml [hunk 1]
```diff
@@ -37,6 +39,8 @@ data:
     gateway.recover_after_time: ${RECOVER_AFTER_TIME:5m}
     gateway.recover_after_master_nodes: ${RECOVER_AFTER_MASTER_NODES:2}
     gateway.recover_after_data_nodes: ${RECOVER_AFTER_DATA_NODES:1}
+    
+{{ toYaml .Values.cluster.config | indent 4 }}
```
Comment (unguiculus) [inline incubator/elasticsearch/templates/elasticsearch-configmap.yaml]: This adds `null` to the config if empty, which causes pods to crash on startup.

**Pull request merged**